platforms/platforms.go

```diff
@@ -231,6 +231,20 @@ func Parse(specifier string) (specs.Platform, error) {
 	return specs.Platform{}, errors.Wrapf(errdefs.ErrInvalidArgument, "%q: cannot parse platform specifier", specifier)
 }
 
+// ParseArray parses an array of specifiers and converts them into an array of specs.Platform
+func ParseArray(specifiers []string) ([]specs.Platform, error) {
```

Comment:
What was this needed for?

Reply:
From what I recall, I believe this was so that users could specify which platform they wanted to perform encryption/decryption for. @stefanberger, could you verify this?

Reply:
`ctr` uses this function here: https://github.com/stefanberger/containerd/blob/encryption_code_plus_ctr.pr/cmd/ctr/commands/images/crypt_utils.go#L308

Reply:
... it enables encrypting for one or multiple platforms of user choice.

Reply:
I think we can move this loop outside in `ctr` and remove this function?

Reply:
@crosbymichael Keep it or move it into ctr ?

Reply:
I'd say keep as is for now

Reply:
@crosbymichael It's do-able so it would depend if you think that this API would be helpful in general, if not we will just apply the following patch:

```
diff --git a/cmd/ctr/commands/images/crypt_utils.go b/cmd/ctr/commands/images/crypt_utils.go
index d1c4b79..0891738 100644
--- a/cmd/ctr/commands/images/crypt_utils.go
+++ b/cmd/ctr/commands/images/crypt_utils.go
@@ -238,7 +238,7 @@ func cryptImage(client *containerd.Client, ctx gocontext.Context, name, newName
                return images.Image{}, err
        }

-       pl, err := platforms.ParseArray(platformList)
+       pl, err := parsePlatformArray(platformList)
        if err != nil {
                return images.Image{}, err
        }
@@ -305,7 +305,7 @@ func getImageLayerInfos(client *containerd.Client, ctx gocontext.Context, name s
                return nil, nil, err
        }

-       pl, err := platforms.ParseArray(platformList)
+       pl, err := parsePlatformArray(platformList)
        if err != nil {
                return nil, nil, err
        }
@@ -403,3 +403,17 @@ func CreateDcParameters(context *cli.Context, descs []ocispec.Descriptor) (map[s

        return dcparameters, nil
 }
+
+// parsePlatformArray parses an array of specifiers and converts them into an array of specs.Platform
+func parsePlatformArray(specifiers []string) ([]ocispec.Platform, error) {
+       var speclist []ocispec.Platform
+
+       for _, specifier := range specifiers {
+               spec, err := platforms.Parse(specifier)
+               if err != nil {
+                       return []ocispec.Platform{}, err
+               }
+               speclist = append(speclist, spec)
+       }
+       return speclist, nil
+}
diff --git a/platforms/platforms.go b/platforms/platforms.go
index 5b3a6a2..2c2cc11 100644
...skipping...
[s

        return dcparameters, nil
 }
+
+// parsePlatformArray parses an array of specifiers and converts them into an array of specs.Platform
+func parsePlatformArray(specifiers []string) ([]ocispec.Platform, error) {
+       var speclist []ocispec.Platform
+
+       for _, specifier := range specifiers {
+               spec, err := platforms.Parse(specifier)
+               if err != nil {
+                       return []ocispec.Platform{}, err
+               }
+               speclist = append(speclist, spec)
+       }
+       return speclist, nil
+}
diff --git a/platforms/platforms.go b/platforms/platforms.go
index 5b3a6a2..2c2cc11 100644
--- a/platforms/platforms.go
+++ b/platforms/platforms.go
@@ -231,20 +231,6 @@ func Parse(specifier string) (specs.Platform, error) {
        return specs.Platform{}, errors.Wrapf(errdefs.ErrInvalidArgument, "%q: cannot parse platform specifier", specifier)
 }

-// ParseArray parses an array of specifiers and converts them into an array of specs.Platform
-func ParseArray(specifiers []string) ([]specs.Platform, error) {
-       var speclist []specs.Platform
-
-       for _, specifier := range specifiers {
-               spec, err := Parse(specifier)
-               if err != nil {
-                       return []specs.Platform{}, err
-               }
-               speclist = append(speclist, spec)
-       }
-       return speclist, nil
-}
-
 // MustParse is like Parses but panics if the specifier cannot be parsed.
 // Simplifies initialization of global variables.
 func MustParse(specifier string) specs.Platform {
...skipping...
[s

        return dcparameters, nil
 }
+
+// parsePlatformArray parses an array of specifiers and converts them into an array of specs.Platform
+func parsePlatformArray(specifiers []string) ([]ocispec.Platform, error) {
+       var speclist []ocispec.Platform
+
+       for _, specifier := range specifiers {
+               spec, err := platforms.Parse(specifier)
+               if err != nil {
+                       return []ocispec.Platform{}, err
+               }
+               speclist = append(speclist, spec)
+       }
+       return speclist, nil
+}
diff --git a/platforms/platforms.go b/platforms/platforms.go
index 5b3a6a2..2c2cc11 100644
--- a/platforms/platforms.go
+++ b/platforms/platforms.go
@@ -231,20 +231,6 @@ func Parse(specifier string) (specs.Platform, error) {
        return specs.Platform{}, errors.Wrapf(errdefs.ErrInvalidArgument, "%q: cannot parse platform specifier", specifier)
 }

-// ParseArray parses an array of specifiers and converts them into an array of specs.Platform
-func ParseArray(specifiers []string) ([]specs.Platform, error) {
-       var speclist []specs.Platform
-
-       for _, specifier := range specifiers {
-               spec, err := Parse(specifier)
-               if err != nil {
-                       return []specs.Platform{}, err
-               }
-               speclist = append(speclist, spec)
-       }
-       return speclist, nil
-}
-
 // MustParse is like Parses but panics if the specifier cannot be parsed.
 // Simplifies initialization of global variables.
 func MustParse(specifier string) specs.Platform {
...skipping...
diff --git a/cmd/ctr/commands/images/crypt_utils.go b/cmd/ctr/commands/images/crypt_utils.go
index d1c4b79..0891738 100644
--- a/cmd/ctr/commands/images/crypt_utils.go
+++ b/cmd/ctr/commands/images/crypt_utils.go
@@ -238,7 +238,7 @@ func cryptImage(client *containerd.Client, ctx gocontext.Context, name, newName
                return images.Image{}, err
        }

-       pl, err := platforms.ParseArray(platformList)
+       pl, err := parsePlatformArray(platformList)
        if err != nil {
                return images.Image{}, err
        }
@@ -305,7 +305,7 @@ func getImageLayerInfos(client *containerd.Client, ctx gocontext.Context, name
s
                return nil, nil, err
        }

-       pl, err := platforms.ParseArray(platformList)
+       pl, err := parsePlatformArray(platformList)
        if err != nil {
                return nil, nil, err
        }
@@ -403,3 +403,17 @@ func CreateDcParameters(context *cli.Context, descs []ocispec.Descriptor) (map
[s

        return dcparameters, nil
 }
+
+// parsePlatformArray parses an array of specifiers and converts them into an array of specs.Platform
+func parsePlatformArray(specifiers []string) ([]ocispec.Platform, error) {
+       var speclist []ocispec.Platform
+
+       for _, specifier := range specifiers {
+               spec, err := platforms.Parse(specifier)
+               if err != nil {
+                       return []ocispec.Platform{}, err
+               }
+               speclist = append(speclist, spec)
+       }
+       return speclist, nil
+}
diff --git a/platforms/platforms.go b/platforms/platforms.go
index 5b3a6a2..2c2cc11 100644```

Reply:
@lumjjb I like that better to keep the core packages with a smaller interface